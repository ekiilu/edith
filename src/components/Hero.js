import React from "react"

import { makeStyles } from "@material-ui/core/styles"

import Typography from "@material-ui/core/Typography"

const useStyles = makeStyles(theme => ({
  root: {
    marginBottom: theme.spacing(10),
    [theme.breakpoints.up("md")]: {
      textAlign: "center",
      marginLeft: theme.spacing(10),
      marginRight: theme.spacing(10),
    },
  },
  description: {
    fontSize: "20px",
    fontWeight: 300,
  },
}))

/**
 * Hero is used to display important information front and center.
 */
export default props => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <Typography variant="h1">{props.title}</Typography>
      <Typography className={classes.description} paragraph>
        {props.description}
      </Typography>
    </div>
  )
}
