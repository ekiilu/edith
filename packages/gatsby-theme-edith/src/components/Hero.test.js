import React from "react"
import { render } from "@testing-library/react"

import Hero from "./Hero"

describe("Hero", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<Hero title="Foo" description="Bar" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
