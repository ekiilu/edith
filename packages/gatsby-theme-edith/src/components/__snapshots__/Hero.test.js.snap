// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero renders correctly 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1"
    >
      Foo
    </h1>
    <p
      class="MuiTypography-root makeStyles-description-2 MuiTypography-body1 MuiTypography-paragraph"
    >
      Bar
    </p>
  </div>
</DocumentFragment>
`;
